code/modules/integrated_electronics/subtypes/power.dm

```diff
@@ -40,7 +40,7 @@
 	var/atom/movable/AM = get_pin_data_as_type(IC_INPUT, 1, /atom/movable)
 	if(!AM)
 		return FALSE
-	if(istype(AM, /obj/item/gun/energy))
+	if(istype(AM, /obj/item/gun/energy) && !istype(AM.loc,/obj/item/integrated_circuit/manipulation/weapon_firing)) // Can only charge guns inside of the assembly's weapon firing mechanism
```

Comment:
well, this check wasnt there at all before #37424 , so... but is there a way to check if a gun isnt meant to be charged?

Reply:
actually nevermind, i found out the can_charge variable

Reply:
thats because this feature was not developed properly and since then it has been realized fully and needs correcting. Another aspect is this will probably need approval to a degree from @KorPhaeron on how well it can charge weapons or how this should be handled now.

That is why I had to add the check of which you are adding a second check to.